src/main.go

```diff
@@ -22,7 +22,7 @@ func NewTemporalCloudProvider(temporalCloudAdapter ITemporalCloudAdapter, config
 
 	err := temporalCloudAdapter.Flags().Parse(os.Args)
 	if err != nil {
-		return nil, fmt.Errorf("unable to parse flags: %v", err)
```

Comment:
The error message wording was changed from 'unable to parse flags' to 'failed to parse flags' for consistency. Consider defining common error messages as constants if they are reused.